include/stronk/unit.h

```diff
@@ -3,32 +3,19 @@
 #include <utility>
 
 #include <stronk/stronk.h>
+#include <stronk/utilities/dimensions.h>
 #include <stronk/utilities/macros.h>
-#include <stronk/utilities/type_list.h>
 
 namespace twig
 {
 
-namespace details
-{
-struct is_unit_tag
-{};
-struct is_identity_unit_tag
-{};
-}  // namespace details
-
-template<typename T>
-concept unit_list_like = std::same_as<typename T::has_unit_lists, std::true_type>;
-
-template<typename T>
-concept identity_unit_like =
-    unit_list_like<T> && std::same_as<typename T::unit_like_tag, details::is_identity_unit_tag>;
+// Concepts
 
 template<typename T>
-concept unit_like = unit_list_like<T> && std::same_as<typename T::unit_like_tag, details::is_unit_tag>;
+concept unit_like = stronk_like<T> && requires { T::dimensions; };
 
 template<typename T>
-concept pure_unit_like = unit_like<T> && !std::same_as<typename T::pure_t, void>;
+concept identity_unit_like = unit_like<T> && requires { T::dimensions::empty(); };
```

Comment:
Does this require an empty() function to be present, or does it actually respect what it returns?